### PR TITLE
wip: first pulse and init state

### PR DESCRIPTION
**Pull request by rizlik130**

### CODE REVIEW

File: ge.h [hunk 1]
```diff
@@ -131,12 +147,20 @@ struct ge {
     /* Registers for special conditions and exceptions */
     uint8_t rFI[7];
 
+    /* 7 bits */
+    uint8_t ffFI;
```
Comment (biappi-bux) [inline ge.h:151]: there are a few registers that have a non 8/16 number of bits.. what do you think of having some tyedefs like `geuint7_t` `geuint9_t` etc (or similar naming)? even not in the context of this PR?

/cc @danielinux

Comment (rizlik130) [reply ge.h:150]: I think it's a good idea.  If we really want to stretch this we may also include if the type refers to a Flip Flop, a register, a knot, or a signal. Even though I think prefixes are already sufficient for that.

Comment (biappi) [reply ge.h:150]: oh great! i think FFs / registers / knot is a bit too extra mile, unless we need some more complex operations and we want to turn those into structs!

**Review by biappi — APPROVED** [review]

File: tests/init.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#include <check.h>
+
+#include "../ge.h"
+
+START_TEST(initialitiation_state)
```
Comment (danielinux) [inline tests/init.c:5]: spelling

File: ge.c [hunk 3]
```diff
@@ -27,23 +28,47 @@ void ge_clear(struct ge * ge)
...
 /// Emulate the press of the "load" button in the console
-void ge_load(struct ge * ge)
+int ge_load(struct ge *ge, uint8_t *program, uint8_t size)
 {
-    // not emulated
+    if (program == NULL && size != 0)
+        return -1;
+
+    /* When pressing LOAD button, AINI is set. If AINI is set, the state 80
+     * (initialitiation) goes to state c8, starting the loading of the program
```
Comment (danielinux) [inline ge.c:42]: check spelling "initialization"

File: tests/init.c (hunk 2 above)
Comment (biappi) [reply tests/init.c:5]: “Initialitiation” is spelt like this in the original docs… I also kept it like that, do we want to “correct” it, or be more faithful to the manuals?

Comment (rizlik130) [reply tests/init.c:5]: yeah that's also the reason I kept it. I like sticking to the manual

File: ge.c (hunk 3 above)
Comment (rizlik130) [reply ge.c:42]: see above

**Pull request merged**